tests/integration_tests/vectorstores/test_redis.py

```diff
@@ -58,3 +58,37 @@ def test_redis_add_texts_to_existing() -> None:
     output = docsearch.similarity_search("foo", k=2)
     assert output == TEST_RESULT
     assert drop(TEST_INDEX_NAME)
+
+
+class TestRedisDistanceMetrics:
+    """Test using different distance metrics for new indices."""
```

Comment:
Hello, @EandrewJones! :wave:  This is awesome! 

How difficult would it be to expand the test to `k=2` and have a meaningful difference for the 2nd retrieved result between the distance metrics?

The main concern is that with `k=1`, the test doesn't verify that the distance metric works correctly, only that the vector store can be instantiated with the given distance metric value. So a future refactor, could accidentally break this feature.

Reply:
Sure, thing. I can put something together with `k=2` for each metric showing the outputs and scores are as expected.

Though if the distance metrics don't work correctly that would imply an issue with the RedisStack implementation. We cannot directly address that on the langchain side apart from maybe excluding metrics that don't work correctly.

Reply:
Got it--By ensuring the expected results for `k=2` are unique to each metric, then we can guard against someone accidentally re-hardcoding the distance metric somewhere.

Will implement.

Reply:
@eyurtsev Any chance you can explain to me why  if I add an explicit `pytest.failure()` into an integration test, it still passes?

Reply:
@eyurtsev I've updated the test to include assertions unique to each distance metric by leveraging the scores produced for the second document when `k=2`.

Simple but it should cover the above concern you raise.